core/src/Revolution/modNamespace.php

```diff
@@ -157,4 +180,44 @@ public function findPolicy($context = '')
 
         return $policy;
     }
+
+    public static function getExtrasNamespaces(modX $modx): array
+    {
+        $namespaceList = [];
+
+        $c = $modx->newQuery(modTransportPackage::class);
+        $c->select([
+            'name' => 'DISTINCT SUBSTRING_INDEX(`signature`,"-",1)'
+        ]);
+        $namespaces = $modx->getIterator(modTransportPackage::class, $c);
```

Comment:
Shouldn't this be fetching from modNamespace instead of modTransportPackage? 🤔

Reply:
I don't think so, as the purpose of this method is to just get the Extras-related namespaces (and not others, like custom ones).

Reply:
I think @Mark-H is correct. MODX namespaces are namespaces, regardless if they are part of an extra or the core. There's no such thing as an Extras-related namespace based on the signature of a transport package.

Reply:
The thing is I'm guessing you're both just looking at the code and not the context of what it's being used for. No, there's not literally an Extra-specific type of namespace, but I have derived that distinction (based on installed TPs) for purposes of identifying and sorting in specific grids. For example, I often create namespaces that are a home for particular types of customization and transformations. The method we're discussing is how I'm able to identify the "Creator" as "User" instead of "Core" or "Extra" (shown in the few grids that have that column). 

Am I wrong in thinking that every installed Extra does and must have its own unique namespace? I can't think of any I've installed that don't.

Reply:
You are wrong, yes. There is no requirement for an extra to have a unique namespace, nor is there any requirement preventing them from having half-a-dozen namespaces installed from a single transport package. And there is certainly no requirement that the namespace match the transport package (or extra) name.

I honestly have no idea what this Creator as User thing is about or how that is related to MODX namespaces.

Reply:
@smg6511 is this aspect of the PR necessary or do you tihnk it could be pulled for a future PR/improvement? It seems like it could be deferred. We really want to get this merged when we can, but if feels like this piece is optional and needs some review.

Reply:
Alright guys, I pulled this logic for now and will add it to a follow-up PR. (It affected only two grids.) At that point, we can debate the "how" of what I'm wanting to achieve, and I'll demonstrate its value via a screen capture video.